File: a2oj/2017-09-08/10110.py
MAX_N = 65536
es_primo = [True] * (MAX_N + 1)
primos = list()

def marcar_multiplos(step, max_n):
    global es_primo
    for i in range(step, max_n, step):
        es_primo[i] = False

def criba(max_n):
    global es_primo, primos
    primos.append(1)
    for i in range(2, max_n + 1):
        if es_primo[i]:
            primos.append(i)
            marcar_multiplos(i, max_n + 1)
    return primos

primos = criba(MAX_N)

T = int(input())
while T != 0:
    root = T ** .5
    apagada = False
    for i in primos:
        if i > root:
            break
        if T % i == 0:
            apagada = not apagada
    if apagada:
        print("no")
    else:
        print("yes")
    T = int(input())

File: a2oj/2017-09-08/10110.cpp
#include <cmath>
#include <vector>
#include <iostream>

using namespace std;

#define MAX_N 65536
bool es_primo[MAX_N+1] = {true};
vector<int> primos;

void marcar_multiplos(int step, int max_n){
	for(int i=step; i<=max_n; i+=step){
		es_primo[i] = false;
	}
}

void criba(int max_n){
	primos.push_back(1);
	for(int i=2; i<=max_n; i++){
		if(es_primo[i]){
			primos.push_back(i);
			marcar_multiplos(i, max_n);
		}
	}
}

int main(){
	criba(MAX_N);
	int T;
	cin >> T;
	while(T!=0){
		int root = sqrt(T)+1;
		bool apagada = false;
		cout << "primos len " << primos.size() << endl;
		for(int p : primos){
			if(p>root) {
				cout << "break: p=" << p << ", root=" << root << endl;
				break;
			}
			if((T%p)==0) {
				cout << T << " % " << p << " = " << T%p << endl;
				apagada = not apagada;
			}
		}
		if(apagada)
			cout << "no" << endl;
		else
			cout << "yes" << endl;
		cin >> T;
	}
	return 0;
}

